app/models/video.rb

```diff
@@ -0,0 +1,34 @@
+class Video
+  attr_accessor :url
+
+  def initialize(url)
+    self.url = url
+  end
+
+  def twitch?
+    URI.parse(url).host.match?(/^(?:www\.)?(?:twitch\.tv)$/i)
+  end
+
+  def youtube?
+    /^(?:https?:\/\/)?(?:www\.)?youtu(?:be\.com|\.be)\/.+/i.match?(url)
+  end
+
+  def id
+    if youtube?
+      # https://www.youtube.com/watch?v=asdf1234&feature=related
+      # https://youtube.com/asdf1234
```

Comment:
As far as I'm aware this is not a format for ID's, but for channels.

Reply:
Seems plausible. I'll see if there are any possible video URLs that don't use the `v` query string parameter. A quick glances makes it look like that's not the case, so that would simplify the URL check and `video_id` lookup.